packages/react-scripts/scripts/build.js

```diff
@@ -142,6 +142,11 @@ function build(previousSizeMap) {
       process.exit(1);
     }
 
+    if (process.env.CI && stats.compilation.warnings.length) {
+     printErrors('Failed to compile. Note, the build has crashed because it is being run with the environment variable CI set to true. In this mode the build crashes when any warnings are encountered.', stats.compilation.warnings);
```

Comment:
Good comment! We wouldn't want this message to cause any confusion. I will rephrase it:)